File: design-patterns/csharp/bridge/Rectangle.cs
namespace Bridge
{
    public class Rectangle : Shape
    {
        private readonly float width;
        private readonly float height;

        public Rectangle(IRenderer renderer, float width, float height) : base(renderer)
        {
            this.width = width;
            this.height = height;
        }

        public override void Draw()
        {
            renderer.RenderRectangle(width, height);
        }
    }
} 

File: design-patterns/golang/bridge/rectangle.go
package bridge

// Rectangle represents a rectangle shape
type Rectangle struct {
	*BaseShape
	width  float64
	height float64
}

// NewRectangle creates a new Rectangle
func NewRectangle(renderer Renderer, width, height float64) *Rectangle {
	return &Rectangle{
		BaseShape: NewBaseShape(renderer),
		width:     width,
		height:    height,
	}
}

// Draw implements the Shape interface
func (r *Rectangle) Draw() {
	r.renderer.RenderRectangle(r.width, r.height)
}
